fhir-search/src/main/java/com/ibm/fhir/search/sort/Sort.java

```diff
@@ -81,72 +85,77 @@ public static Direction fromValue(String value) {
-            String sortParmValue, boolean lenient) throws Exception {
+    public void parseSortParameter(String resourceTypeName, FHIRSearchContext context, String sortParmValue) throws Exception {
 
         for (String sortParmCode : sortParmValue.split(",")) {
-            // Each parameter now includes the direction.
-            Sort.Direction sortDirection = Direction.fromValue(sortParmCode);
-
-            // In the case, where we are DECREASING, we need to treat the CODE in a special
-            // way by stripping the directional indicator.
-            if (Sort.Direction.DECREASING.compareTo(sortDirection) == 0) {
-                sortParmCode = sortParmCode.substring(1);
-            }
+            try {
+                // Each parameter now includes the direction.
+                Sort.Direction sortDirection = Direction.fromValue(sortParmCode);
+
+                // In the case, where we are DECREASING, we need to treat the CODE in a special
+                // way by stripping the directional indicator.
+                if (Sort.Direction.DECREASING.compareTo(sortDirection) == 0) {
+                    sortParmCode = sortParmCode.substring(1);
+                }
 
-            // Per the FHIR spec, the _sort parameter value is a search parameter. We need to determine what
-            // type of search parameter.
-            SearchParameter sortParmProxy = SearchUtil.getSearchParameter(resourceTypeName, sortParmCode);
-            if (!isUndefinedOrLenient(resourceTypeName, sortParmCode, sortParmProxy, lenient)) {
-                SearchConstants.Type sortParmType =
-                        SearchConstants.Type.fromValue(sortParmProxy.getType().getValue());
-                SortParameter sortParm = new SortParameter(sortParmCode, sortParmType, sortDirection);
+                // Per the FHIR spec, the _sort parameter value is a search parameter. We need to determine what
+                // type of search parameter.
+                SearchParameter sortParmProxy = SearchUtil.getSearchParameter(resourceTypeName, sortParmCode);
+                if (!isUndefinedOrLenient(resourceTypeName, sortParmCode, sortParmProxy, context)) {
+                    SearchConstants.Type sortParmType =
+                            SearchConstants.Type.fromValue(sortParmProxy.getType().getValue());
+                    SortParameter sortParm = new SortParameter(sortParmCode, sortParmType, sortDirection);
 
-                checkSystemLevel(resourceTypeName, sortParm.getCode());
+                    checkSystemLevel(resourceTypeName, sortParm.getCode(), context);
 
-                context.getSortParameters().add(sortParm);
+                    context.getSortParameters().add(sortParm);
+                }
+            } catch (FHIRSearchException se) {
+                // If we're in lenient mode and there was an issue parsing the _sort parameter value then log and move on to the next one.
+                if (context.isLenient()) {
```

Comment:
Where is this set?

Reply:
It is set at the top of SearchUtil.parseQueryParameters(...), which calls SearchUtil.parseSearchResultParameter(...), which calls this method.